packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardTitle.base.tsx

```diff
@@ -95,72 +100,84 @@ export class DocumentCardTitleBase extends BaseComponent<IDocumentCardTitleProps
+        );
+        // Used to calculate the number of chars that should be truncated.
+        const overFlowRate: number = scrollWidth / (parseInt(style.width, 10) * lines);
+
+        if (overFlowRate > 1) {
+          const truncatedLength: number = originalTitle.length / overFlowRate - 3 /** Saved for separator */;
+          return this.setState({
+            truncatedTitleFirstPiece: originalTitle.slice(0, truncatedLength / 2),
+            truncatedTitleSecondPiece: originalTitle.slice(originalTitle.length - truncatedLength / 2),
+            clientWidth,
+            needMeasurement: false
+          });
+        }
+      }
     }
+
+    return this.setState({ needMeasurement: false });
   };
 
-  private _shrinkTitle(): void {
-    if (this._doesTitleOverflow()) {
-      const { truncatedTitleFirstPiece, truncatedTitleSecondPiece } = this.state;
-      this._isTruncated = true;
+  private _shrinkTitle: () => void = () => {
+    const { truncatedTitleFirstPiece, truncatedTitleSecondPiece } = this.state;
+    if (truncatedTitleFirstPiece && truncatedTitleSecondPiece) {
+      const titleElement = this._titleElement.current;
+
+      if (!titleElement) {
+        return false;
+      }
 
-      if (truncatedTitleFirstPiece || truncatedTitleSecondPiece) {
+      if (
+        titleElement.scrollHeight > titleElement.clientHeight + TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD ||
+        titleElement.scrollWidth > titleElement.clientWidth
+      ) {
         this.setState({
-          truncatedTitleFirstPiece: truncatedTitleFirstPiece!.slice(0, truncatedTitleFirstPiece!.length - 1),
-          truncatedTitleSecondPiece: truncatedTitleSecondPiece!.slice(1)
+          truncatedTitleFirstPiece: truncatedTitleFirstPiece.slice(0, truncatedTitleFirstPiece.length - 1),
+          truncatedTitleSecondPiece: truncatedTitleSecondPiece.slice(1)
         });
       }
     }
-  }
-
-  private _doesTitleOverflow(): boolean {
-    const titleElement = this._titleElement.current;
-
-    if (!titleElement) {
-      return false;
-    }
-
-    return (
-      titleElement.scrollHeight > titleElement.clientHeight + TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD ||
-      titleElement.scrollWidth > titleElement.clientWidth
-    );
-  }
+  };
 
   private _updateTruncation(): void {
-    // Only update truncation if the title's size has changed since the last time we truncated
-    if (this._titleElement.current && this._titleElement.current.clientWidth !== this._truncatedTitleAtWidth) {
-      // Throttle truncation so that it doesn't happen during a window resize
-      clearTimeout(this._scrollTimerId);
-      this._scrollTimerId = this._async.setTimeout(this._startTruncation.bind(this, this.props), 250);
-    }
+    this._async.requestAnimationFrame(() => {
+      // Only update truncation if the title's size has changed since the last time we truncated
+      if (this._titleElement.current) {
+        const clientWidth: number = this._titleElement.current.clientWidth;
```

Comment:
The height can change without the width changing, right? Should that be another check?

Reply:
I don't think so, `widt`h will decide the `height`, if `width` is not changing, the `height` won't change, as the text is not changed.